### PR TITLE
Update user version on close

### LINKED ISSUE #4947: Add user version updating to DSA VersionService.close and version close endpoint
Version.close and version close endpoint should accept a new optional parameter for updating the user version.

The options should be:
* Nothing (default)
* Create new user version
* Update last user version to this object version. (If update last user version to this object version and there is no last user version create a new user version).

Please add the parameter to the openapi.

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made? 🤔

Closes #4947 

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Comment by jcoyne** [comment]
@aaron-collier can you squash these commits?

File: app/services/version_service.rb [hunk 1]
```diff
@@ -121,21 +123,39 @@ def can_open?(assume_accessioned: false)
...
+    repository_object = RepositoryObject.find_by!(external_identifier: druid)
+    current_closed_version = repository_object&.last_closed_version
+
     # Default to creating accessionWF when calling close_version
     workflow_client.close_version(druid:,
                                   version: version.to_s,
                                   create_accession_wf: start_accession)
 
-    RepositoryObject.find_by!(external_identifier: druid).close_version!(description:)
+    repository_object&.close_version!(description:)
```
Comment (jcoyne) [inline app/services/version_service.rb:148]: I don't think we need this safe-nav here.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/version_service.rb [hunk 2]
```diff
@@ -121,21 +123,39 @@ def can_open?(assume_accessioned: false)
...
+
+    raise ArgumentError, "user_version must be one of #{user_version_options.join(', ')}" unless user_version_options.include?(user_version)
+
     ObjectVersion.update_current_version(druid:, description:) if description
 
     ensure_closeable!
 
+    # Get the currently closed version before we close the head version
+    repository_object = RepositoryObject.find_by!(external_identifier: druid)
+    current_closed_version = repository_object&.last_closed_version
```
Comment (jcoyne) [inline app/services/version_service.rb:141]: I don't think we need this safe-nav either, do we?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/version_service.rb (hunk 1 above)
Comment (aaron-collier) [reply app/services/version_service.rb:148]: Probably not. I'll update. (same with below)

**Comment by aaron-collier** [comment]
@jcoyne squashed :-)

**Review by jcoyne — APPROVED** [review]

**Pull request merged**